node/src/sub_lib/stream_key.rs

```diff
@@ -97,83 +95,42 @@ type HashType = [u8; sha1::DIGEST_LENGTH];
 #[cfg(test)]
 mod tests {
     use super::*;
-    use std::str::FromStr;
-
-    #[test]
-    fn matching_keys_and_matching_addrs_make_matching_stream_keys() {
-        let key = PublicKey::new(&b"These are the times"[..]);
-        let addr = SocketAddr::from_str("2.3.4.5:6789").unwrap();
-
-        let one = StreamKey::new(key.clone(), addr);
-        let another = StreamKey::new(key, addr);
-
-        assert_eq!(one, another);
-    }
-
-    #[test]
-    fn matching_keys_and_mismatched_addrs_make_mismatched_stream_keys() {
-        let key = PublicKey::new(&b"These are the times"[..]);
-        let one_addr = SocketAddr::from_str("2.3.4.5:6789").unwrap();
-        let another_addr = SocketAddr::from_str("3.4.5.6:6789").unwrap();
-
-        let one = StreamKey::new(key.clone(), one_addr);
-        let another = StreamKey::new(key, another_addr);
-
-        assert_ne!(one, another);
-    }
-
-    #[test]
-    fn matching_keys_and_mismatched_port_numbers_make_mismatched_stream_keys() {
-        let key = PublicKey::new(&b"These are the times"[..]);
-        let one_addr = SocketAddr::from_str("3.4.5.6:6789").unwrap();
-        let another_addr = SocketAddr::from_str("3.4.5.6:7890").unwrap();
-
-        let one = StreamKey::new(key.clone(), one_addr);
-        let another = StreamKey::new(key, another_addr);
-
-        assert_ne!(one, another);
-    }
+    use crate::test_utils::make_meaningful_stream_key;
+    use std::collections::HashSet;
 
     #[test]
-    fn mismatched_keys_and_matching_addrs_make_mismatched_stream_keys() {
-        let one_key = PublicKey::new(&b"These are the times"[..]);
-        let another_key = PublicKey::new(&b"that try men's souls"[..]);
-        let addr = SocketAddr::from_str("2.3.4.5:6789").unwrap();
+    fn stream_keys_are_unique() {
+        let mut stream_keys_set = HashSet::new();
 
-        let one = StreamKey::new(one_key.clone(), addr);
-        let another = StreamKey::new(another_key, addr);
+        for i in 1..=1_000_000 {
```

Comment:
I think a million is a little high. Probably 100 or 1000 is fine. The stream key only has to be unique within a set of streams to or from the same entity (browser, Node, server), and you're not going to see more than 1000 streams in a set like that.